feat: Add exclude_function_args parameter to reduce transaction response sizes

alexthuth opened the pull request:
## Description

As an API client developer, I want to control the size of transaction responses by excluding function arguments when they are not needed. This is necessary because transaction responses can vary dramatically in size (5KB to 50KB+) due to complex contract function arguments, forcing clients to use conservative pagination and resulting in 3-5x more API calls than optimal.

This pull request adds the `exclude_function_args` query parameter to transaction endpoints. When set to `true`, the API omits the `function_args` array from contract call transactions, significantly reducing response size while maintaining all other data.

**What was changed:**
- Added `ExcludeFunctionArgsParamSchema` parameter schema 
- Modified `parseContractCallMetadata` to conditionally exclude function args
- Threaded parameter through all transaction parsing functions
- Updated transaction endpoints: `GET /extended/v1/tx`, `GET /extended/v1/tx/:tx_id`
- Updated address transaction endpoints
- Added comprehensive test coverage

**How this impacts application developers:**
- Enables predictable response sizes for better pagination planning
- Reduces bandwidth usage and improves response times
- 100% backward compatible (defaults to `false`)
- Allows more efficient bulk transaction processing

**Example usage:**
```bash
# Standard response with function_args (larger)
GET /extended/v1/tx/0x123

# Optimized response without function_args (smaller, predictable)
GET /extended/v1/tx/0x123?exclude_function_args=true
```

For implementation details refer to issue #2310

## Type of Change
- [x] New feature
- [ ] Bug fix
- [ ] API reference/documentation update
- [ ] Other

## Does this introduce a breaking change?
No breaking changes. The parameter is optional and defaults to `false`, preserving existing behavior. All existing API calls continue to work identically.

## Are documentation updates required?
- [ ] Link to documentation updates: OpenAPI schema is automatically updated via TypeBox

## Testing information

**Testing is required** for this change.

**Test coverage includes:**
1. Parameter validation (boolean, optional, defaults)
2. Function args exclusion for contract call transactions  
3. Preservation of function args when parameter is false/omitted
4. Non-contract transactions unaffected
5. End-to-end integration tests for both transaction endpoints

**Affected code paths:**
- Transaction parsing pipeline (`parseDbTx`, `parseDbMempoolTx`)
- Contract call metadata processing (`parseContractCallMetadata`)
- Database controller transaction retrieval
- Transaction and address API endpoints

**Testing steps:**
1. Run `npm run test:api` to verify all tests pass
2. Test transaction endpoints with and without the parameter
3. Verify response sizes are reduced when parameter is `true`
4. Confirm backward compatibility with existing clients

## Checklist
- [x] Code is commented where needed
- [x] Unit test coverage for new or modified code paths
- [x] `npm run test` passes
- [x] Changelog is updated
- [x] Tag 1 of @rafaelcr or @zone117x for review

rafaelcr commented on CHANGELOG.md:
CHANGELOG changes are auto-generated whenever a new release is published

alexthuth commented:
With the two most recent commits, we changed:

* Require excludeFunctionArgs (boolean, no default) – all call-sites now pass false.
* Deleted CHANGELOG.md (removed 8.11.0 stub entirely).
* Trimmed description in params.ts.
* Removed obsolete parse-contract-call-metadata.test.ts.

Touched files:
* src/api/controllers/db-controller.ts, four route files, params.ts, +2 deletions (test & changelog).

rafaelcr commented on CHANGELOG.md:
Apologies, I should've explained better how this file works. This file is edited by `semantic-release` automatically whenever a new API release is published. As such, it should stay as part of the repository because we want to keep its (automatic) edit history. In other words, the file was deleted with your latest commit but it should still remain in the repo unchanged. 

Can you revert this file to how it was before? You can do that with
```
git checkout develop -- CHANGELOG.md
```

Sorry for the confusion on this!

rafaelcr commented on src/api/controllers/db-controller.ts:
```diff
@@ -878,6 +879,7 @@ interface GetTxsWithEventsArgs extends GetTxsArgs {
 interface GetTxsArgs {
   txIds: string[];
   includeUnanchored: boolean;
+  excludeFunctionArgs?: boolean;
```
```suggestion
  excludeFunctionArgs: boolean;
```

rafaelcr commented on src/api/controllers/db-controller.ts:
```diff
@@ -864,6 +864,7 @@ export async function getRosettaTransactionFromDataStore(
 interface GetTxArgs {
   txId: string;
   includeUnanchored: boolean;
+  excludeFunctionArgs?: boolean;
```
```suggestion
  excludeFunctionArgs: boolean;
```

rafaelcr reviewed (commented):
Just one last batch of comments but it's looking good!
